### PR TITLE
Feature/testing params interface

### PR DESCRIPTION
**Pull request by lucas-daher**
#### What is the purpose of this pull request?
Create a testing parameters generic interface, so we can create different testing parameters classes to be used in different AB tests. Currently, only one class is created, for the test using conversion as main metric. Also starts saving test type in vbase and adds an optional parameter to `InitializeAbTest`.

#### What problem is this solving?
This makes way for adding revenue as test metric by allowing the creation of different classes implementing `ITestingParameters`.

#### How should this be manually tested?
2 tests:

Test 1:
- Link this version of the app
- Start an AB test
- Check if everything works (all the tester routes and Colossus periodic update route)
- Unlink the app
- Link a new one from master
- Check if everything still works

Test 2:
- Start an AB test
- Check if everything works (all the tester routes and Colossus periodic update route)
- Link this version of the app
- Check if everything still works
 
#### Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: node/typings/testingParameters.ts [hunk 1]
```diff
@@ -69,12 +81,12 @@ const MapInitialParameters = (workspaces: ABTestWorkspace[]): Map<string, ABTest
     }
     return map
 }
-const UnmapParameters = (mapParameters: Map<string, ABTestParameters>): ABTestParameters[] => {
-    const parameters: ABTestParameters[] = []
-    for (const parameter of mapParameters.values()) {
-        parameters.push(parameter)
+
+export function createTestingParameters(testType: TestType): ITestingParameters {
+    if (testType == TestType.conversion) {
```
Comment (ruybrito106) [inline node/typings/testingParameters.ts]: **nit**: this seems like a good opportunity for a ternary :awesome:

File: node/typings/testingParameters.ts [hunk 2]
```diff
@@ -1,12 +1,32 @@
...
+    Update(workspacesData: Map<string, WorkspaceData>): void;
+
+    UpdateWithFixedParameters(proportion: number): void;
+}
+
+export class TestingParametersConversion implements ITestingParameters{
     private parameters: Map<string, ABTestParameters>
 
-    constructor(testingWorkspaces: ABTestWorkspace[]) {
+    constructor() {
```
Comment (ruybrito106) [inline node/typings/testingParameters.ts]: it looks like the constructor and the `Init` methods have the same purpose, right? doesnt it make sense to join them? if there are multiple use cases with different initilization params, maybe you could make them optional.

File: node/typings/testingParameters.ts [hunk 3]
```diff
@@ -1,12 +1,32 @@
...
+
+    Get(): Map<string, ABTestParameters>;
+
+    Add(workspaceName: string): void;
+
+    Remove(workspaceName: string): void;
+
+    Update(workspacesData: Map<string, WorkspaceData>): void;
+
+    UpdateWithFixedParameters(proportion: number): void;
```
Comment (pedro823) [inline node/typings/testingParameters.ts]: *nit*: `camelCase`

File: node/typings/testingParameters.ts (hunk 1 above)
Comment (pedro823) [reply node/typings/testingParameters.ts]: even though the program's returning the same thing inside the `if` and outside?

File: node/typings/testingParameters.ts [hunk 4]
```diff
@@ -69,12 +81,12 @@ const MapInitialParameters = (workspaces: ABTestWorkspace[]): Map<string, ABTest
     }
     return map
 }
-const UnmapParameters = (mapParameters: Map<string, ABTestParameters>): ABTestParameters[] => {
-    const parameters: ABTestParameters[] = []
-    for (const parameter of mapParameters.values()) {
-        parameters.push(parameter)
+
+export function createTestingParameters(testType: TestType): ITestingParameters {
```
Comment (pedro823) [inline node/typings/testingParameters.ts]: *nit*: `export const createTestingParameters = (testType: TestType): ITestingParameters => new TestingParametersConversion()`

**Comment by pedro823** [comment]
oops

File: node/typings/testingParameters.ts (hunk 1 above)
Comment (ruybrito106) [reply node/typings/testingParameters.ts]: 😕

Comment (lucas-daher) [reply node/typings/testingParameters.ts]: I know this part of the code doesn't make much sense now, but other `if`s will be added in the near future (at least one more).

**Pull request merged**